### PR TITLE
Migrate Microsoft.Bot.Builder.Dialogs.Adaptive.Tests to xUnit

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 4093

## Description
This PR migrates all [Microsoft.Bot.Builder.Dialogs.Adaptive](https://github.com/microsoft/botbuilder-dotnet/tree/main/tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests) tests from **MSTest** to **xUnit**.

## Specific Changes
- Replaced `MSTest` packages with `xUnit` in the `.csproj`.
- Changed `[TestClass]` to `[CollectionDefinition]`.
- Changed `[TestMethod]` to `[Fact]`.
- Changed `Assert.AreEqual` to `Assert.Equal`.
- Changed `Assert.IsNotNull` to `Assert.NotNull`.
- Changed `Assert.IsNull` to `Assert.Null`.
- Added `[assembly: TestFramework("Microsoft.Bot.Builder.Integration.ApplicationInsights.Core.Tests.Startup", "Microsoft.Bot.Builder.Dialogs.Adaptive.Tests")]` to the `Startup` class to replace `[AssemblyInitialize]`.
- Implemented `ResourceExplorerFixture` class to share the `ResourceExplorer` among tests.
- Implemented `ConfigurationFixture` class to share the `Configuration` among tests.
- Improved tests Assertions.

## Testing
In the following image there are the passing tests.
image

### CODE REVIEW

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/AdaptiveDialogTests.cs [hunk 1]
```diff
@@ -306,17 +296,17 @@ protected override async Task<DialogTurnResult> OnBeginDialogAsync(DialogContext
...
             var ds = new TestAdaptiveDialog();
             var version1 = ds.GetInternalVersion_Test();
-            Assert.IsNotNull(version1);
+            Assert.NotNull(version1);
 
             var ds2 = new TestAdaptiveDialog();
-            var version2 = ds.GetInternalVersion_Test();
-            Assert.IsNotNull(version2);
-            Assert.AreEqual(version1, version2, "Same configuration should give same version");
+            var version2 = ds2.GetInternalVersion_Test();
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/AdaptiveDialogTests.cs:307]: I think we should keep this as it was.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ConfigurationFixture.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
+﻿using System;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ConfigurationFixture.cs:1]: Don't forget to add the copyright header in every new file.

File: tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ResourceExplorerFixture.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
```
Comment (ceciliaavila) [inline tests/Microsoft.Bot.Builder.Dialogs.Adaptive.Tests/ResourceExplorerFixture.cs:1]: Add the copyright header please.

**Review by ceciliaavila — CHANGES_REQUESTED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by santgr11 — APPROVED** [review]
lgtm

**Review by ceciliaavila — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Comment by denscollo** [comment]
Promoted

**Pull request closed without merging**